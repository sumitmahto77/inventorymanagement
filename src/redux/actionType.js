// // export const GET_USERS ="GET_USERS";
// // export const ADD_USER = "ADD_USER";

// export const LOGIN_START = "LOGIN_START";
// export const LOGIN_SUCCESS = "LOGIN_SUCCESS";
// export const LOGIN_FAIL = "LOGIN_FAIL";

// export const LOGOUT_USER = "LOGOUT_USER";

// export const REGISTER_START = "REGISTER_START";
// export const REGISTER_SUCCESS = "REGISTER_SUCCESS";
// export const REGISTER_FAIL = "LOGIN_FAIL";



