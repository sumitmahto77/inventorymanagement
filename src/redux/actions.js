import * as types from './actionType';
import axios from 'axios';

// const getUsers = (users) => ({
//     type: types.GET_USERS,
//     payload: users,
// });

// const userAdded = () =>({
//     type:types.ADD_USER,

// });

// export const loadUsers = () => {
//     return function(dispatch){
//         axios.get("http://localhost:4000/users")
//         .then((response) => {
//             console.log("response", response);
//             dispatch(getUsers(response.data));
//         })
//         .catch((error) => console.log(error));
//     };
// };

// export const addUser = (user) => {
//     return function (dispatch){
//         axios
//             .post("http://localhost:4000/users",user)
//             .then((response) =>{
//                 console.log("response",response);
//                 dispatch(userAdded());
//             })
//             .catch((error) => console.log(error));
//     };
// };


const loginStart = () => ({
    type : types.LOGIN_START,
});

const loginSuccess = (token) => ({
    type : types.LOGIN_SUCCESS,
    payload : token,
});

const loginFail = (error) => ({
    type : types.LOGIN_FAIL,
    payload : error,
});

const registerStart = () => ({
    type : types.REGISTER_START,
});

const registerSuccess = (token) => ({
    type : types.REGISTER_SUCCESS,
    payload : token,
});

const registerFail = (error) => ({
    type : types.REGISTER_FAIL,
    payload : error,
});

export const logout = () =>({
    type : types.LOGOUT_USER,

})


export const loginInitiate = (email,password) => {
    return function(dispatch){
        dispatch(loginStart());
        axios
        .post("http://localhost:4000/login",{
            email,
            password,
        })
        .then((response) => {
            console.log("response", response);
            dispatch(loginSuccess(response.data.access_token))
            
        })
        .catch((error) => dispatch(loginFail(error.response.data.message)));
    };
};
export const registerInitiate = (user) => {
    return function(dispatch){
        dispatch(registerStart());
        axios
        .post("http://localhost:4000/users",user)
        .then((response) => {
            console.log("response", response);
            dispatch(registerSuccess(response.data))
            
        })
        .catch((error) => dispatch(registerFail(error.response.data.message)));
    };
};
