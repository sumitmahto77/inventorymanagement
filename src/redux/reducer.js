import * as types from './actionType';

const initialAuthState ={
    user : null,
    loading : false,
    error : null,
    token : null
};

const initialProdState = {
    products : [],
    loading : true,
};

export const authReducer = (state = initialAuthState, action) => {
    switch (action.type){
        case types.LOGIN_START:
        case types.REGISTER_START:
            return{
                ...state,
                loading : true,
            };
        case types.LOGIN_SUCCESS:
        case types.REGISTER_SUCCESS:
            return{
                ...state,
                loading: false,
                user :action.payload.user,
                token : action.payload.accessToken,
            };
        case types.LOGIN_FAIL:
        case types.REGISTER_FAIL:
            return{
                ...state,
                loading :false,
                error : action.payload,
            };

        case types.LOGOUT_USER:
            return{
                ...state,
                user : null,
                token :null,
            };

        default:
            return state;
    }
};

export const prodReducer =(state=initialProdState,action) => {
    switch (action.type){
        case types.GET_PRODUCTS:
            return {
                ...state,
                products : action.payload,
                loading: false,
            };
        case types.ADD_PRODUCT:
            return{
                ...state
            }
        case types.DELETE_PRODUCT:
            return{
                ...state,
                loading :false,
            }

        default:
            return state;
    }

}
