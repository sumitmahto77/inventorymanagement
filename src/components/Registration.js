import React,{useState} from "react";
import styled from "styled-components";
import {mobile} from '../responsive';
import {useDispatch} from 'react-redux';
import { useHistory} from 'react-router-dom';
import { registerInitiate } from "../redux/actions";


const Container = styled.div`
  width: 100vw;
  height: 100vh;
  background: linear-gradient(
      rgba(255, 255, 255, 0.5),
      rgba(255, 255, 255, 0.5)
    ),
    url("https://images.pexels.com/photos/6984661/pexels-photo-6984661.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=650&w=940")
      center;
  background-size: cover;
  display: flex;
  align-items: center;
  justify-content: center;
`;

const Wrapper = styled.div`
  width: 40%;
  padding: 20px;
  background-color: white;
  ${mobile({ width: "75%" })}

`;

const Title = styled.h1`
  font-size: 24px;
  font-weight: 300;
`;

const Form = styled.form`
  display: flex;
  flex-wrap: wrap;
`;

const Input = styled.input`
  flex: 1;
  min-width: 40%;
  margin: 20px 10px 0px 0px;
  padding: 10px;
`;

const Agreement = styled.span`
  font-size: 12px;
  margin: 20px 0px;
`;

const Button = styled.button`
  width: 40%;
  border: none;
  padding: 15px 20px;
  background-color: teal;
  color: white;
  cursor: pointer;
`;

const Registration = () => {
  const [error, setError] = useState("");
  let dispatch = useDispatch();
  let history = useHistory();
 
  const [formData, setFormData] = useState({
    firstName: "",
    lastName: "",
    location: "",
    email: "",
    phone: "",
    password: ""
  })

  const {firstName,lastName,location,email,phone,password} = formData;

  function handleSubmit(e) {
    e.preventDefault();
    if (!firstName || !lastName || !location || !email || !phone || !password) {
      setError("Please input all fields");
    }else{
      dispatch(registerInitiate(formData));
      history.push("/");
      setError("");
    }
    
}

function handleChange(e) {
    setFormData({...formData, [e.target.name] : e.target.value})
}

  return (
    <Container>
      <Wrapper>
        <Title>CREATE AN ACCOUNT</Title>

        <Form onSubmit={e=>handleSubmit(e)}>
          <Input placeholder="first name" name="firstName" value={firstName} type="text" onChange={e=>handleChange(e)}/>
          <Input placeholder="last name" name="lastName" value={lastName} type="text" onChange={e=>handleChange(e)}/>
          <Input placeholder="location" name="location" value={location} type="text" onChange={e=>handleChange(e)}/>
          <Input placeholder="email" name="email" value={email} type="email" onChange={e=>handleChange(e)}/>
          <Input placeholder="phone number" name="phone" value={phone} type="text" onChange={e=>handleChange(e)}/>
          <Input placeholder="password" name="password" value={password} type="password" onChange={e=>handleChange(e)}/>
          <Agreement>
            By creating an account, I consent to the processing of my personal
            data in accordance with the <b>PRIVACY POLICY</b>
          </Agreement>
          <Button type="submit">CREATE</Button>
        </Form>
      </Wrapper>
    </Container>
  );
};

export default Registration;
