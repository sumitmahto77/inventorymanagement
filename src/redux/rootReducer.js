import {combineReducers} from "redux";
import {persistReducer} from "redux-persist";
import storage from "redux-persist/lib/storage";
import {authReducer,prodReducer} from "./reducer";




// const rootReducer = combineReducers({
//     users: usersReducers
// });

// export default rootReducer; 



const persistConfig = {
    key : "root",
    storage,
    whitelist: ["auth","prod"]
}


const rootReducer = combineReducers({
    auth : authReducer,
    prod : prodReducer,
});

export default persistReducer(persistConfig, rootReducer); 

