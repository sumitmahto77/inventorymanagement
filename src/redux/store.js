import {createStore, applyMiddleware} from "redux";
import {composeWithDevTools} from 'redux-devtools-extension';
import logger from 'redux-logger';
import reduxThunk from 'redux-thunk';
import {persistStore} from "redux-persist";
import rootReducer from './rootReducer';

const middlewares =[reduxThunk];

if (process.env.NODE_ENV === "development"){
    middlewares.push(logger)
}

export const store = createStore(rootReducer, composeWithDevTools(applyMiddleware(...middlewares)) );

export const persistor = persistStore(store); 


