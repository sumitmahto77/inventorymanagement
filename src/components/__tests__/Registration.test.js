import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import {store} from '../../redux/store';
import Registration from '../Registration';
import {render, screen} from '@testing-library/react';
import '@testing-library/jest-dom';

describe('Registration component Snapshot', () =>{
    test('Renders correctly', () => {
        const tree = renderer.create(
            <Provider store={store}>
            <Registration/>
            </Provider>
        );
        expect(tree).toMatchSnapshot();
    });    
});

describe ('Registration page rendering of elements', () =>{
    beforeEach(() => {
        render (<Provider store={store}><Registration/></Provider>)
    });

    test('renders correct heading for Registration page', () =>{
        const heading = screen.getByRole('heading',{level:1});
        expect(heading).toHaveTextContent('CREATE AN ACCOUNT');
    });
});