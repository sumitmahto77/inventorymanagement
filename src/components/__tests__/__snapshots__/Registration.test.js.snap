// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Registration component Snapshot Renders correctly 1`] = `
<div
  className="sc-bczRLJ gAJGBz"
>
  <div
    className="sc-gsnTZi bhLUiU"
  >
    <h1
      className="sc-dkzDqf cBSyNJ"
    >
      CREATE AN ACCOUNT
    </h1>
    <form
      className="sc-hKMtZM ePPhml"
      onSubmit={[Function]}
    >
      <input
        className="sc-eCYdqJ bxJwlw"
        name="firstName"
        onChange={[Function]}
        placeholder="first name"
        type="text"
        value=""
      />
      <input
        className="sc-eCYdqJ bxJwlw"
        name="lastName"
        onChange={[Function]}
        placeholder="last name"
        type="text"
        value=""
      />
      <input
        className="sc-eCYdqJ bxJwlw"
        name="location"
        onChange={[Function]}
        placeholder="location"
        type="text"
        value=""
      />
      <input
        className="sc-eCYdqJ bxJwlw"
        name="email"
        onChange={[Function]}
        placeholder="email"
        type="email"
        value=""
      />
      <input
        className="sc-eCYdqJ bxJwlw"
        name="phone"
        onChange={[Function]}
        placeholder="phone number"
        type="text"
        value=""
      />
      <input
        className="sc-eCYdqJ bxJwlw"
        name="password"
        onChange={[Function]}
        placeholder="password"
        type="password"
        value=""
      />
      <span
        className="sc-jSMfEi eVQROJ"
      >
        By creating an account, I consent to the processing of my personal data in accordance with the 
        <b>
          PRIVACY POLICY
        </b>
      </span>
      <button
        className="sc-gKXOVf dJiKET"
        type="submit"
      >
        CREATE
      </button>
    </form>
  </div>
</div>
`;
